channels/postgres/channels.go

```diff
@@ -696,8 +696,11 @@ direct_groups_with_subgroup AS (
 		"groups" g ON g.id = gr.entity_id
 	JOIN
 		groups_role_actions all_actions ON all_actions.role_id = grm.role_id
+	LEFT JOIN "groups" g2
+		ON g2.path <@ g.path AND nlevel(g2.path) = nlevel(g.path) + 1
 	WHERE
-		grm.member_id = '%s'
+		g2.path IS NULL
+		AND grm.member_id = '%s'
 		AND g.domain_id = '%s'
 		AND gra."action" LIKE 'subgroup_channel%%'
```

Comment:
Please move `g2.path IS NULL` to last in in `WHERE`
```suggestion
	LEFT JOIN "groups" g2
		ON g2.path <@ g.path AND nlevel(g2.path) = nlevel(g.path) + 1
	WHERE
		grm.member_id = '%s'
		AND g.domain_id = '%s'
		AND gra."action" LIKE 'subgroup_channel%%'
		AND g2.path IS NULL
```